wrappers/core18.go

```diff
@@ -661,18 +661,9 @@ func undoSnapdDbusActivationOnCore() error {
 	return err
 }
 
-func snapdDesktopFileNames(s *snap.Info) ([]string, error) {
-	paths, err := filepath.Glob(filepath.Join(s.MountDir(), "usr/share/applications/*.desktop"))
-	if err != nil {
-		return nil, err
-	}
-
-	fileNames := make([]string, len(paths))
-	for i, path := range paths {
-		fileNames[i] = filepath.Base(path)
-	}
-
-	return fileNames, nil
+var snapdDesktopFileNames = []string{
+	"io.snapcraft.SessionAgent.desktop",
+	"snap-handle-link.desktop",
```

Comment:
I might be completely wrong here, please correct me. I am seeing `snap-userd-autostart.desktop` under `data/desktop`, do we also need this file or is it irrelevant?

Reply:
It gets installed to `/etc/xdg/autostart/snap-userd-autostart.desktop`, and is used quite differently to the desktop files in `/usr/share/applications`.

It's something we probably need to sort out for Core Desktop at some point, but it'll require more than just installing the desktop file: the code that runs autostart files there is part of a snap, and the command in this file can only function unconfined.

Reply:
got it, thank you for clarifying!